tensorflow/contrib/seq2seq/python/ops/layers.py

```diff
@@ -20,15 +20,185 @@
 from __future__ import division
 from __future__ import print_function
 
+from tensorflow.python.ops import tensor_array_ops
 from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import variable_scope as vs
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import control_flow_ops
+from tensorflow.python.ops import rnn
 
+from tensorflow.contrib.layers import fully_connected
+
+from functools import partial
 
 __all__ = ["rnn_decoder",
            "rnn_decoder_attention"]
 
+"""Used to project encoder state in `rnn_decoder`"""
+encoder_projection = partial(fully_connected, activation_fn=math_ops.tanh)
 
-def rnn_decoder(*args, **kwargs):
-  pass
+
+def rnn_decoder(cell, decoder_inputs, initial_state,
+                sequence_length, decoder_fn,
+                encoder_projection=encoder_projection,
+                parallel_iterations=None, swap_memory=False,
+                time_major=False, scope=None):
+  """RNN decoder for a sequence-to-sequence model specified by RNNCell 'cell'.
```

Comment:
change quotes to backquotes: `cell`; here and everywhere you refer to objects (rnn_decoder below, the namespace dynamic_rnn, etc)